### PR TITLE
DSEGOG-269 Make waveform thumbnail size configurable

### PR DESCRIPTION
**Pull request by MRichards99**
This PR makes the waveform thumbnail size configurable - this has always been an option in the file but never used in the codebase. I've also added a new config option called `line_width` as it seems sensible that this is configurable.

I've added tests to ensure this feature is monitored from now on - there are tests for both image thumbnails as waveform thumbnails.

To test this PR locally, make sure you update your config file and run the `ingest_echo_data.py` again. A diff of the example config file can be found at https://github.com/ral-facilities/operationsgateway-api/pull/133/commits/fe86286a59c4c40f2ef35d8565c68170fd77fe8f

### CODE REVIEW

File: test/images/test_image.py [hunk 1]
```diff
@@ -63,6 +63,41 @@ def test_create_thumbnail(self):
...
+    @pytest.mark.parametrize(
+        "image_size, config_thumbnail_size, expected_thumbnail_size",
+        [
+            pytest.param((300, 300), (50, 50), (50, 50), id="50x50 thumbnail"),
+            pytest.param((400, 300), (60, 80), (60, 80), id="60x80 thumbnail"),
+            pytest.param(
+                (300, 300),
+                (75, 100),
+                (75, 75),
+                id="75x100 thumbnail (square image)",
```
Comment (kevinphippsstfc) [inline test/images/test_image.py:75]: This should probably be labelled '75x75 thumbnail (square image)'. The currently labelling confused me a bit.

**Review by kevinphippsstfc — CHANGES_REQUESTED** [review]
Looks pretty good. Just a few minor comments on the things we discussed earlier today.

File: operationsgateway_api/src/records/waveform.py [hunk 2]
```diff
@@ -77,19 +78,25 @@ def _create_thumbnail_plot(self, buffer) -> None:
...
+            linewidth=Config.config.waveforms.line_width,
         )
+        # Disables all axis decorations
         plt.axis("off")
+        # Removes the frame around the plot
         plt.box(False)
 
+        # Setting bbox_inches="tight" and pad_inches=0 removes padding around figure
+        # to make best use of the limited pixels available in a thumbnail
         plt.savefig(buffer, format="PNG", bbox_inches="tight", pad_inches=0, dpi=130)
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply operationsgateway_api/src/records/waveform.py:104]: Added a comment in 74992688d4bd21ab0f4e52ad18e8689493a84244

File: test/images/test_image.py [hunk 3]
```diff
@@ -63,6 +63,41 @@ def test_create_thumbnail(self):
         # the reason only 0s are being asserted is because this is checking the hash of
         # a purely black 300x300 square created in the test_image above
 
+    @pytest.mark.parametrize(
+        "image_size, config_thumbnail_size, expected_thumbnail_size",
+        [
+            pytest.param((300, 300), (50, 50), (50, 50), id="50x50 thumbnail"),
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply test/images/test_image.py:73]: Added a comment to explain this in 74992688d4bd21ab0f4e52ad18e8689493a84244

File: test/images/test_image.py [hunk 4]
```diff
@@ -63,6 +63,41 @@ def test_create_thumbnail(self):
...
+            pytest.param((400, 300), (60, 80), (60, 80), id="60x80 thumbnail"),
+            pytest.param(
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply test/images/test_image.py:88]: Added in 74992688d4bd21ab0f4e52ad18e8689493a84244 - I've labelled each of the test cases to say what the image orientation should be (I've done this for the equivalent tests for waveform thumbnails as well)

File: test/images/test_image.py (hunk 1 above)
Comment (MRichards99) [reply test/images/test_image.py]: Good point, on reflection I can see how that might be confusing. Changed in 74992688d4bd21ab0f4e52ad18e8689493a84244

**Pull request merged**